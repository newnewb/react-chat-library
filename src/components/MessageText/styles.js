// @ts-check

const textStyle = {
	fontSize: 16,
	lineHeight: 20,
	marginTop: 5,
	marginBottom: 5,
	marginLeft: 10,
	marginRight: 10
};

const styles = {
	left: {
		text: {
			color: 'black',
			...textStyle
		},
		link: {
			color: 'black',
			textDecorationLine: 'underline'
		}
	},
	right: {
		text: {
			color: 'white',
			...textStyle
		},
		link: {
			color: 'white',
			textDecorationLine: 'underline'
		}
	}
};

export default styles;
