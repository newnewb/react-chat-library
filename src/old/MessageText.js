import PropTypes from 'prop-types';
import React from 'react';
import { StyleSheet, Text, View, ViewPropTypes } from 'react-native-web';

// import ParsedText from 'react-native-parsed-text';

export default class MessageText extends React.Component {
	constructor(props) {
		super(props);
	}

	shouldComponentUpdate(nextProps) {
		return this.props.currentMessage.text !== nextProps.currentMessage.text;
	}

	render() {
		const linkStyle = StyleSheet.flatten([
			styles[this.props.position].link,
			this.props.linkStyle[this.props.position]
		]);
		return (
			<View
				style={[
					styles[this.props.position].container,
					this.props.containerStyle[this.props.position]
				]}
			>
				{/* <ParsedText
					style={[
						styles[this.props.position].text,
						this.props.textStyle[this.props.position],
						this.props.customTextStyle
					]}
					parse={[...this.props.parsePatterns(linkStyle)]}
					childrenProps={{ ...this.props.textProps }}
				> */}
				{this.props.currentMessage.text}
				{/* </ParsedText> */}
			</View>
		);
	}
}

const textStyle = {
	fontSize: 16,
	lineHeight: 20,
	marginTop: 5,
	marginBottom: 5,
	marginLeft: 10,
	marginRight: 10
};

const styles = {
	left: StyleSheet.create({
		container: {},
		text: {
			color: 'black',
			...textStyle
		},
		link: {
			color: 'black',
			textDecorationLine: 'underline'
		}
	}),
	right: StyleSheet.create({
		container: {},
		text: {
			color: 'white',
			...textStyle
		},
		link: {
			color: 'white',
			textDecorationLine: 'underline'
		}
	})
};

MessageText.contextTypes = {
	actionSheet: PropTypes.func
};

MessageText.defaultProps = {
	position: 'left',
	currentMessage: {
		text: ''
	},
	containerStyle: {},
	textStyle: {},
	linkStyle: {},
	customTextStyle: {},
	textProps: {},
	parsePatterns: () => []
};

MessageText.propTypes = {
	position: PropTypes.oneOf(['left', 'right']),
	currentMessage: PropTypes.object,
	containerStyle: PropTypes.shape({
		left: ViewPropTypes.style,
		right: ViewPropTypes.style
	}),
	textStyle: PropTypes.shape({
		left: Text.propTypes.style,
		right: Text.propTypes.style
	}),
	linkStyle: PropTypes.shape({
		left: Text.propTypes.style,
		right: Text.propTypes.style
	}),
	parsePatterns: PropTypes.func,
	textProps: PropTypes.object,
	customTextStyle: Text.propTypes.style
};
