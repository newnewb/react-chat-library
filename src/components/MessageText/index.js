// @ts-check

// Packages
import React from 'react';
import { linkifier } from 'react-linkifier';

// Ours
import { CSS } from '../../utils';
import styles from './styles';

/**
 * @typedef Props
 * @property {string} [text]
 * @property {string} [position]
 * @property {object} [textStyle]
 * @property {object} [linkStyle]
 * @param {Props} props
 */
const MessageText = props => (
	<div
		style={CSS([styles[props.position].text, props.textStyle[props.position]])}
	>
		{linkifier(props.text, {
			target: '_blank',
			style: CSS([styles[props.position].link, props.linkStyle])
		})}
	</div>
);

/** @type Props */
MessageText.defaultProps = {
	position: 'left',
	text: '',
	textStyle: {},
	linkStyle: {}
};

export default MessageText;
