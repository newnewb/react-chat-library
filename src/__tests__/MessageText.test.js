import React from 'react';
import renderer from 'react-test-renderer';

import MessageText from '../components/MessageText';

it('should render <MessageText /> and compare with snapshot', () => {
	const component = renderer.create(<MessageText />);
	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});
