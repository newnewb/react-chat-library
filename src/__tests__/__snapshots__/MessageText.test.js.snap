// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <MessageText /> and compare with snapshot 1`] = `
<div
  style={
    Object {
      "color": "black",
      "fontSize": 16,
      "lineHeight": 20,
      "marginBottom": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 5,
    }
  }
/>
`;
